### PR TITLE
DIG-1355: Sonchau/UUID with fk

### PR DESCRIPTION
**Pull request by SonQBChau**
#### What's new:

- All tables will have FKs in the same manner as currently, but these FKs will be uuids
submitter_<object>_ids will be stored in each object as a text field.
- UUIDs will remain internal to katsu and not be exposed via any API.
- UUIDs will be generated by katsu when data is ingested
- The combination of (program_id, submitter_*_id) will have a UNIQUE constraint in the database (for donor, specimen, sample, treatment and follow-up ids)

#### Potentially Breaking changes:

- DonorWithClinicalData no longer has filters. Can only retrieve 1 specific object by providing the `program_id` and `submitter_*_id`.
- The structure of the DonorWithClinicalData request has changed and is now [`"/v2/authorized/donor_with_clinical_data/program/{program_id}/donor/{donor_id}"`](https://github.com/CanDIG/katsu/blob/2a7f44a9bf153215e4839427483c626aaee158a0/chord_metadata_service/mohpackets/docs/schema.json#L515)
- Ingest APIs no longer accept a list of object.
- Discovery query no longer has filters. This will be addressed in the next patch
- New data ingest script for development using 
`python chord_metadata_service/mohpackets/data/data_loader.py`

#### Package changes:

- **NEW** django-ninja for both API and schema + filter
- **NEW** orjson faster JSON package for renderer since we use a lot of json
- **REMOVE** drf, replaced by ninja, used serializer which is slower and hard to customize
- **REMOVE** django-filter, replaced by ninja, only work with drf
- **REMOVE** drf-spectacular, replaced by ninja, only work with drf. This used to generate schema based on the serializer to yml file. Since the new ninja only generate json, we need to convert it to yml

#### How to test:
- Due to significant changes, it is recommended to drop the database and re-create
- `pip install -r requirements/local.txt`
- `tox`

#### Related tickets:
[DIG-1389](https://candig.atlassian.net/browse/DIG-1389)
[DIG-1312](https://candig.atlassian.net/browse/DIG-1312)

[DIG-1389]: https://candig.atlassian.net/browse/DIG-1389?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mshadbolt** [comment]
@SonQBChau are you able to add a line to the summary at the top that explains the change to the different package (including the package name) and how it might affect others - change from yaml to json schema, thank you!

**Comment by SonQBChau** [comment]
I found a new bug with authorization in docker, temporary take it down

File: .github/workflows/update-schema.yaml [hunk 1]
```diff
@@ -37,17 +41,17 @@ jobs:
...
 
-      - name: Generate new schema.yml
-        run: python manage.py spectacular --file ./chord_metadata_service/mohpackets/docs/schema.yml --validate --fail-on-warn
+      - name: Generate new schema.json
+        run: python manage.py export_openapi_schema --api chord_metadata_service.mohpackets.apis.core.api | python -m json.tool > chord_metadata_service/mohpackets/docs/schema.json
 
       - name: Convert schema to OpenAPI documentation
         run: |
-          npx widdershins ./chord_metadata_service/mohpackets/docs/schema.yml -o ./chord_metadata_service/mohpackets/docs/openapi.md -u ./chord_metadata_service/mohpackets/docs/widdershins/templates/openapi3 -c true --omitHeader true
+          npx widdershins ./chord_metadata_service/mohpackets/docs/schema.json -o ./chord_metadata_service/mohpackets/docs/openapi.md -u ./chord_metadata_service/mohpackets/docs/widdershins/templates/openapi3 -c true --omitHeader true
```
Comment (daisieh) [inline .github/workflows/update-schema.yaml]: Could you add a step to do something like https://stackoverflow.com/questions/38781929/how-to-convert-json-to-yaml-in-javascript, to minimize downstream changes?

**Comment by SonQBChau** [comment]
> @SonQBChau are you able to add a line to the summary at the top that explains the change to the different package (including the package name) and how it might affect others - change from yaml to json schema, thank you!

I added the packages change

**Review by daisieh — CHANGES_REQUESTED** [review]
NestedFollowUpSchema does not have any properties in the schema.yml file.

There are still some typos being generated into the schema.yml/json files:
- primarydiagnosis instead of primary_diagnoses
- hormonetherapies instead of hormone_therapies
- samplesregistrations instead of sample_registrations
I think these are misnamed as properties of their parent NestedWhatever objects?

**Comment by SonQBChau** [comment]
- NestedFollowUpSchema is just a typo, I put Schema instead of ModelSchema so it didn't show up.
- The schema is not being updated is a little complicated. This is a bug through github action. The new schema is generated, but it is not committed yet. However, the commit script can only read the old schema on github repo to get the SHA(which is a requirement to update), which is why the newly generated content is lost. I have to think about it.

**Review by daisieh — APPROVED** [review]
As far as I can tell, this works as expected. Looks good!

**Review by OrdiNeu — APPROVED** [review]
From the data-portal side everything looks fine as far as I can tell, will need some modifications to data portal code to get it working with this but nothing too out-of-the-ordinary :heavy_check_mark: .

**Pull request merged**